ASPNetCoreMastersTodoList.Api/Controllers/ItemsController.cs

```diff
@@ -11,39 +11,82 @@
 
 namespace ASPNetCoreMastersTodoList.Api.Controllers
 {
+    [ApiController]
+    [Route("[controller]")]
     public class ItemsController : Controller
     {
         private readonly ILogger<ItemsController> _logger;
-        private ItemService _itemService;
 
-        public ItemsController(ILogger<ItemsController> logger, ItemService itemService)
+        private IItemService _itemService;
+
+        public ItemsController(ILogger<ItemsController> logger, IItemService itemService)
         {
             _logger = logger;
             _itemService = itemService;
         }
 
         [HttpGet]
-        IEnumerable<string> GetAll(int userId)
+        IActionResult Get()
+        {
+            var result = _itemService.GetAll();
+
+            return Ok(result);
+        }
+
+        [HttpGet]
+        [Route("{itemId}")]
+        IActionResult Get(int itemId)
         {
-            return _itemService.GetAll(userId);
+            var result = _itemService.Get(itemId);
+
+            return Ok(result);
         }
 
         [HttpGet]
-        int Get(int userId)
+        [Route("filterBy")]
+        IActionResult GetByFilters([FromBody] string filters)
         {
-            return 1;
+            var itemFilterDto = new ItemByFilterDTO(filters);
+
+            var result = _itemService.GetAllByFilter(itemFilterDto);
+
+            return Ok(result);
         }
 
         [HttpPost]
-        void Save([FromBody] ItemCreateBindingModel modelObject)
+        IActionResult Post([FromBody] ItemCreateBindingModel itemCreateModel)
+        {
+            var itemDto = new ItemDTO() { Text = itemCreateModel.Text };
+
+            _itemService.Add(itemDto);
+
+            return Ok();
+        }
+
+        [HttpPut]
+        [Route("{itemId}")]
+        IActionResult Put(int itemId, [FromBody] ItemUpdateBindingModel itemUpdateModel)
+        {
+            //where to use itemId?
```

Comment:
In restful design it is more appropriate to use the id from the route. The body of the request shouldn't have an id. No need to change just additional info

Reply:
Thank you for reviewing  - this was updated.